h0rton/h0_inference/h0_posterior.py

```diff
@@ -0,0 +1,269 @@
+
+        offset = np.concatenate([ra_offset, dec_offset], axis=1)
+        return np.linalg.norm(offset, axis=1)
+
+    def reorder_to_tdlmc(self, img_array):
+        """Apply the permutation scheme for reordering the list of ra, dec, and time delays to conform to the order in the TDLMC challenge
+
+        Parameters
+        ----------
+        img_array : array-like
+            array of properties corresponding to the AGN images
+
+        Returns
+        -------
+        array-like
+            `img_array` reordered to the TDLMC order
+        """
+        img_array = np.array(img_array)[self.increasing_dec_i][self.abcd_ordering_i]
+        return img_array
+
+    def get_samples(self, n_samples, seed, exclude_vel_disp=False):
+        """Get samples from the H0Posterior
+
+        Parameters
+        ----------
+        n_samples : int
+        seed : int
+            random seed to use
+        exclude_vel_disp : bool
+            whether to exclude the velocity dispersion likelihood. Default: False
+
+        Returns
+        -------
+        dict
+            keys are `H0_samples` (array of shape (n_samples,)), `H0_weights` (array of shape (n_samples,)), `inferred_vd` (inferred velocity dispersions, array of shape (n_samples,)), `ll_vd` (log likelihood of the inferred velocity dispersions, array of shape (n_samples,)), `inferred_td` (inferred time delays, array of shape (n_samples, n_images)), `ll_td` (log likelihood of the inferred time delays, array of shape (n_samples, n_images))
+
+        """
+        # Seed for reproducibility
+        random.seed(seed)
+        np.random.seed(seed)
+        # Initialize output dict
+        samples_dict = dict(
+                            H0_samples=np.zeros(n_samples),
+                            H0_weights=np.zeros(n_samples),
+                            inferred_vd=np.zeros(n_samples),
+                            ll_vd=np.zeros(n_samples),
+                            inferred_td=np.zeros((n_samples, self.n_img)),
+                            ll_td=np.zeros((n_samples, self.n_img)),
+                            )
+
+        for n in tqdm(range(n_samples)):
+            H0_candidate = self.sample_H0()
+            k_ext = self.sample_kappa_ext()
+            aniso_param = self.sample_aniso_param()
+            # Define cosmology
+            cosmo = FlatLambdaCDM(H0=H0_candidate, Om0=self.Om0, Ob0=self.Ob0)
+            # Tool for getting time delays and velocity dispersions
+            lens_prop = LensProp(self.z_lens, self.z_src, self.kwargs_model, cosmo=cosmo)
+            # Velocity dispersion
+            if exclude_vel_disp:
+                ll_vd = 0.0
+            else:
+                inferred_vd = lens_prop.velocity_dispersion(self.kwargs_lens, r_eff=self.lens_light_R_sersic, R_slit=self.R_slit, dR_slit=self.dR_slit, psf_fwhm=self.psf_fwhm, aniso_param=aniso_param, num_evaluate=5000)
+                ll_vd = log_normal_pdf(inferred_vd,
```

Comment:
do you have log normal errors here or Gaussian ones? (I think the tdlmc has Gaussian errors and I am not sure this is what you are calculating here). If its the natural logarithm of the Gaussian error, it's correct.

Reply:
It's the natural log of the Gaussian error. The naming is confusing. I've switched to `log_gaussian_pdf` and clarified in the parameter description that `mu` and `sigma` are both on a linear scale.